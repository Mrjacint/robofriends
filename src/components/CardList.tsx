import React from 'react';
import Card from './Card';
import { IRobots, IRobot } from '../types';

const CardList: React.FunctionComponent<IRobots> = ({ robots }) => {
  return (
    <div>
      {
        robots.map((user: IRobot, i: number) => {
          return (
            <Card 
              key={i}
              id={user.id}
              name={user.name}
              email={user.email}
              />
          );
        })
      }
    </div>
  );
}

export default CardList;