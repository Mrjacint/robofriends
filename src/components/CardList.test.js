import { shallow } from "enzyme";
import React from "react";
import CardList from "./CardList";

it("expect to render without crashing", () => {
  const mockRobots = [
    {
      id: 1,
      name: "Jhon Doe",
      username: "JhonJhon",
      email: "johon@gmail.com",
    },
  ];

  expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
});
