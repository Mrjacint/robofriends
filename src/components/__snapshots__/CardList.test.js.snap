// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render without crashing 1`] = `
<div>
  <Card
    email="johon@gmail.com"
    id={1}
    key="0"
    name="Jhon Doe"
  />
</div>
`;
